### ISSUE TITLE
new generation login_id

### ISSUE THREAD

**Issue by k-morozov** [k-morozov/App-chat#34]
new generation login_id

**Comment by k-morozov** [comment]
Need to rewrite the login_id generation logic

**Comment by ashishsiyag** [comment]
@k-morozov I read your code and logic you have used for login_id is starting from value 0 and incrementing it as new client logs in, right?

**Comment by k-morozov** [comment]
@ashishsiyag Yes. A new client_id is generated for a new client (starting from 0, increasing by 1). This is a bad solution because when  server is restarting - for all new clients will be generated client_id starting from 0. The solution (as I think):
1. generating id on a 32 bit random number (and check that it's unique).
or .
2. Save the counter information for id in the database.

**Comment by ashishsiyag** [comment]
@k-morozov We can choose a hash function to generate the 32-bit client_id for the new client

**Comment by k-morozov** [comment]
@ashishsiyag variable size only

**Comment by ashishsiyag** [comment]
@k-morozov more of whether the operation need to be O(1) or O(logn) will do? To check whether the key exists we need to look through the previous keys, if we directly store them into database and check whether it exists or not, will take O(logn).

**Comment by k-morozov** [comment]
@ashishsiyag If you can implement complexity O(1) - excellent.
